capstone/src/main/webapp/scripts/profile.js

```diff
@@ -196,36 +202,21 @@ function closeModal(type) {
 
 /** Populate form with user values */
 function populateEditForm(user) {
-  document.getElementById("modal-name-container").innerText = user.name;
+  document.getElementById("modal-name-container").innerText = user.firstName + " " + user.lastName;
   document.getElementById("first").value = user.firstName;
   document.getElementById("last").value = user.lastName;
   document.getElementById("email").value = user.email;
   document.getElementById("phone").value = user.phoneNumber;
   document.getElementById("address").value = user.address;
   document.getElementById("interests").value = user.interests.join(", ");
-}
 
-/** Save updated profile information */
-function saveEdits() {
-  const editForm = document.getElementById("edit-profile");
-  if (editForm.reportValidity()) {
-    const firstName = document.getElementById("first").value;
-    const lastName = document.getElementById("last").value;
-    const email = document.getElementById("email").value;
-    const phoneNumber = document.getElementById("phone").value;
-    const address = document.getElementById("address").value;
-    const interests = document.getElementById("interests").value;
-
-    const params = new URLSearchParams();
-    params.append("first", firstName);
-    params.append("last", lastName);
-    params.append("email", email);
-    params.append("phone", phoneNumber);
-    params.append("address", address);
-    params.append("interests", interests);
-
-    // Send a POST request to the servlet which edits the user profile.
-    fetch("/editProfile", { method: "POST", body: params });
+  // Add user profile picture if previously uploaded 
+  if (user.profilePic != null & user.profilePic != "") {
+    let profilePicDiv = document.getElementById("modal-profile-pic");
+    let imageContent = document.createElement("img");
+    imageContent.className = "profile-pic-img";
+    imageContent.src = "serve?blob-key=" + user.profilePic;
+    profilePicDiv.appendChild(imageContent);
   }
 }
```

Comment:
Are you sure we want to delete this and it's not used elsewhere?

Reply:
^Yes I think deleting saveEdits() might have been a mistake?

Reply:
Oops nevermind, I see you did it in the html instead

Reply:
Interesting, totally missed that. I also didn't know that's how forms needed to work for blobstore, thanks for the explanation!